### PR TITLE
feat: single character shortcut events

### PR DESCRIPTION
**Pull request by hyj1204**
Part of https://github.com/AppFlowy-IO/appflowy-editor/issues/101
This PR is to add single character shortcut events and their test code, including 

-   ⭐️ backtick to code: `abc` to abc.
-   ⭐️ tilde to strikethrough: ~abc~ to abc.
-   ⭐️ single asterisk or single underline to italic: *abc* to abc, _abc_ to abc.

`handleFormatByWrappingWithSingleChar` is used to return the handler(function) we need in `CharacterShortcutEvent`. It can be used for style formatting by any single character. 

Enum `FormatStyleByWrappingWithSingleChar` is to limit the formatting styles. It supports code, italic, strikethrough so far. 

https://user-images.githubusercontent.com/14248245/234732129-5c65f755-081f-4f05-9483-b44a6cb0bdd6.mov

### CODE REVIEW

File: lib/src/editor/editor_component/service/shortcuts/character_shortcut_events.dart [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export 'character_shortcut_events/insert_newline.dart';
 export 'character_shortcut_events/slash_command.dart';
-export 'character_shortcut_events/format_italic.dart';
+export 'character_shortcut_events/single_character_shortcut_events/single_character_shortcut_events.dart';
```
Comment (LucasXu0) [inline lib/src/editor/editor_component/service/shortcuts/character_shortcut_events.dart]: Actually, all the shortcut event based on character is single. So, this file name is not good.

File: lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/single_character_shortcut_events/handle_single_character_format.dart [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import 'package:appflowy_editor/appflowy_editor.dart';
+
+enum SingleCharacterFormatStyle {
```
Comment (LucasXu0) [inline lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/single_character_shortcut_events/handle_single_character_format.dart]: MarkdownSymbol ??

File: lib/src/editor/editor_component/service/shortcuts/character_shortcut_events.dart (hunk 1 above)
Comment (hyj1204) [reply lib/src/editor/editor_component/service/shortcuts/character_shortcut_events.dart]: I renamed the folder to `format_by_wrapping_with_single_char`.

File: lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/single_character_shortcut_events/handle_single_character_format.dart (hunk 2 above)
Comment (hyj1204) [reply lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/single_character_shortcut_events/handle_single_character_format.dart]: I renamed it to `FormatStyleByWrappingWithSingleChar` may make more sense.  It is designed to be used for any style formatting that can be triggered by wrapping text with a single character. Currently, it includes code, italic, and strikethrough styles.

File: lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/format_by_wrapping_with_single_char/handle_format_by_wrapping_with_single_char.dart [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+enum FormatStyleByWrappingWithSingleChar {
+  code,
+  italic,
+  strikethrough,
+}
+
+Future<bool> Function(EditorState) handleFormatByWrappingWithSingleChar({
+  required String char,
+  required FormatStyleByWrappingWithSingleChar formatStyle,
+}) {
```
Comment (LucasXu0) [inline lib/src/editor/editor_component/service/shortcuts/character_shortcut_events/format_by_wrapping_with_single_char/handle_format_by_wrapping_with_single_char.dart]: This function looks a little tricky and confusing. I think you can simplify it by just using editorState as the parameter.

**Pull request merged**